content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1066 @@
+
+First, to set up the `PrismaClient` instance that you'll use to send database queries from the various route handlers. Create a new file named `prisma.js` in the `src` directory:
+
+```terminal copy
+touch src/prisma.js
+```
+
+Now, instantiate `PrismaClient` and export it from the file so you can use it in your route handlers later:
+
+```js copy file=src/prisma.js
+const { PrismaClient } = require('@prisma/client')
+
+const prisma = new PrismaClient()
+
+module.exports = prisma
+```
+
+The imports in our controller files are as follows:
+
+```js file=src/controllers/post.js
+const Post = require('../models/post')
+const User = require('../models/user')
+const Category = require('../models/category')
+```
+
+```js file=src/controllers/user.js
+const Post = require('../models/post')
+const User = require('../models/user')
+```
+
+You'll update the controller imports as you migrate from Mongoose to Prisma:
+
+```js file=src/controllers/post.js
+const prisma = require('../prisma')
+```
+
+```js file=src/controllers/user.js
+const prisma = require('../prisma')
+```
+
+### 4.1. Replacing queries in `GET` requests
+
+The REST API has four routes that accept `GET` requests:
+
+- `/feed?searchString={searchString}&take={take}&skip={skip}`: Return all published posts
+  - Query Parameters (optional):
+    - `searchString`: Filter posts by `title` or `content`
+    - `take`: Specifies how many objects should be returned in the list
+    - `skip`: Specifies how many of the returned objects should be skipped
+- `/post/:id`: Returns a specific post
+- `/authors`: Returns a list of authors
+
+Let's dive into the route handlers that implement these requests.
+
+#### `/feed`
+
+The `/feed` handler is currently implemented as follows:
+
+```js file=src/controllers/post.js
+const feed = async (req, res) => {
+  try {
+    const { searchString, skip, take } = req.query
+
+    const or =
+      searchString !== undefined
+        ? {
+            $or: [
+              { title: { $regex: searchString, $options: 'i' } },
+              { content: { $regex: searchString, $options: 'i' } },
+            ],
+          }
+        : {}
+
+    const feed = await Post.find(
+      {
+        ...or,
+        published: true,
+      },
+      null,
+      {
+        skip,
+        batchSize: take,
+      }
+    )
+      .populate({ path: 'author', model: User })
+      .populate('categories')
+
+    return res.status(200).json(feed)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+Note that each returned `Post` object includes the relation to the `author` and `category` it's associated with. With Mongoose, including the relation is not type-safe. For example, if there was a typo in the relation that is retrieved, your database query would fail only at _runtime_ – the JavaScript compiler does not provide any safety here.
```

Comment:
```suggestion
Note that each returned `Post` object includes the relation to the `author` and `category` with which it is associated. With Mongoose, including the relation is not type-safe. For example, if there was a typo in the relation that is retrieved, your database query would fail only at _runtime_ – the JavaScript compiler does not provide any safety here.
```